### ISSUE TITLE
BBOT 2.0 multiprocessing oopsie

### ISSUE THREAD

**Issue by TheTechromancer** [blacklanternsecurity/bbot#1511]
![image](https://github.com/blacklanternsecurity/bbot/assets/20261699/1d2db415-aa24-4a57-a548-1fd8f81ac123)

somebody forgot to do `if __name__ == "__main__"`

**Comment by TheTechromancer** [comment]
After merging a [fix](https://github.com/blacklanternsecurity/bbot/pull/1514) for this, python3.9 and 3.10 tests are suddenly timing out. The individual tests themselves are finishing properly, but the pytest process never exits.

I ran the tests in a docker container and what I discovered was that there is a "thread leak". It appears that threads are not being killed properly. As the scan continues, more and more threads are spawned, and the number keeps going up. By the end, there are hundreds.

This is a very strange issue. I'm actually not sure where the threads are being spawned, since we are making sure to kill the DNS/Web engine ones when we are finished with them.
